packages/vite/src/node/baseEnvironment.ts

```diff
@@ -5,56 +5,85 @@ import type { Plugin } from './plugin'
 
 export class PartialEnvironment {
   name: string
-  config: ResolvedConfig
-  options: ResolvedEnvironmentOptions
+  getTopLevelConfig(): ResolvedConfig {
+    return this._topLevelConfig
+  }
+
+  config: ResolvedConfig & ResolvedEnvironmentOptions
+
+  // TODO: keep for back-compat with earlier v6 alpha versions
+
+  // options: ResolvedEnvironmentOptions
+
+  _options: ResolvedEnvironmentOptions
+
   logger: Logger
 
+  /**
+   * @internal
+   */
+  _topLevelConfig: ResolvedConfig
+
   constructor(
     name: string,
-    config: ResolvedConfig,
-    options: ResolvedEnvironmentOptions = config.environments[name],
+    topLevelConfig: ResolvedConfig,
+    options: ResolvedEnvironmentOptions = topLevelConfig.environments[name],
   ) {
     this.name = name
-    this.config = config
-    this.options = options
+    this._topLevelConfig = topLevelConfig
+    this._options = options
+    this.config = new Proxy(
+      options as ResolvedConfig & ResolvedEnvironmentOptions,
+      {
+        get: (target, prop: keyof ResolvedConfig) => {
+          if (prop === 'logger') {
+            return this.logger
+          }
+          if (prop in target) {
+            return this._options[prop as keyof ResolvedEnvironmentOptions]
```

Comment:
Does `resolve` fully override default? 🤔

Reply:
Yes, we now made `resolve.alias` per-environment too. I think the only option that doesn't make much sense to use per-environment is `resolve.preserveSymlinks` but I think it doesn't hurt. I think we should have any per environment object option to be fully overriden

Reply:
So if I override only a single property there, the whole `resolve` is now overriden. Sounds like a source of unexpected issues to be honest

Reply:
Oh, this isn't how it works. During the config resolution, the ResolvedConfig will merge the defaults with the specific values in each `config.environments[name]`. So if you only override a single property, that is the only one that is overriden.

Reply:
Ok then 👍